### DISCUSSION TITLE
Upgrade for Angular to V2

### DISCUSSION THREAD

**Discussion by daofabdotcom** [WalletConnect/walletconnect-monorepo#2676] [Developer Support]
Hi there, 

We have followed the steps listed here:- 

https://docs.walletconnect.com/2.0/web/web3modal/html/introduction

However, we are encountering issues in runtime on Wallet Connect Library (as Angular is not typescript). 

![18913b70-a64e-4f24-9420-5bd85c822122](https://github.com/WalletConnect/walletconnect-monorepo/assets/25010428/e951b883-4e85-4e59-899e-134fc54fbb82)
![e2439157-9dee-4562-9641-a774e4444ea4](https://github.com/WalletConnect/walletconnect-monorepo/assets/25010428/92d2095e-d7e5-465e-8a9e-fa02a3c3bf20)

![955f013c-d970-49c3-9b5c-ee36b52ecb5d](https://github.com/WalletConnect/walletconnect-monorepo/assets/25010428/0a9109fe-d2d1-4ca6-9ac9-8c5cf0e76850)

Kindly advise.

**Comment by glitch-txs** [comment]
Hi could you share a minimal reproducible example?

**Comment by finessevanes** [comment]
As @glitch-txs asked, could you share a minimal reproducible example and include the SDK version you're using.

**Comment by sabetghadam** [comment]
i use angular 15 and walletconnect 1.8 since days ago complete walletconnect not working an dapp is pause use , why since 6 month ago for angular do not make example or guid for migrate ? its really bad choise

**Comment by damon-leong** [comment]
Hi, we have this issue due to the strict type checking, is there any guidance or documentation to migrate to angular v16 for Walletconnect 2.0? or can tell us what is the best approach if there is no official support for angular?

**Reply by glitch-txs** [reply]
Hi, which library are you using? All web libraries work with Angular excepting the React one.

**Comment by glitch-txs** [comment]
If you're using wagmi make sure you have typescript on strict mode and v5
https://wagmi.sh/core/typescript

**Reply by damon-leong** [reply]
Hi we have issue missing module because cannot find it on wed3modal, is this possible to manual add in or how should resolve it?

Error: node_modules/@web3modal/html/dist/_types/src/client.d.ts:17:45 - error TS2307: Cannot find module 'packages/core/dist/_types/src/types/controllerTypes' or its corresponding type declarations.

setDefaultChain: (selectedChain: import("packages/core/dist/_types/src/types/controllerTypes").Chain | undefined) => void;

**Reply by glitch-txs** [reply]
Are you using typescript? Can you share your ts.config file?

**Reply by csurbier** [reply]
Hi,

Here is a fresh example with Angular 16, typescript 5 showing the problems  : 
https://github.com/csurbier/walletconnectv2-angular.git

I'm stuck too for months now trying to figure out how to make WalletConnect V2 working with Angular !

Thanks

**Reply by glitch-txs** [reply]
@csurbier you need to remove @walletconnect/modal-sign-html dependency and install @web3modal/html instead. I was able to run your project successfully 👍

**Reply by csurbier** [reply]
Hi, thanks. Yes i forgot to update this post. Few minutes later. I have solved the problem and updated my git repository readme.md to let other users known. Thanks to wagmi GitHub issues. 
 Need to add an option in the tsconfig file to be able to compile : skipLibCheck to true.

Thanks anyway for having a look.
Christophe
